TestV2WatchKeyInDir-2 fail

=== RUN TestV2WatchKeyInDir-2
panic: HTTP body JSON parse error: unexpected end of JSON input: 
goroutine 112949 [running]:
github.com/coreos/etcd/integration.(*testHttpClient).ReadBodyJSON(0xc20a9aeea0, 0xc2154c6120, 0x35)
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/v2_http_kv_test.go:995 +0x2a7
github.com/coreos/etcd/integration.func·003()
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/v2_http_kv_test.go:878 +0x1de
created by github.com/coreos/etcd/integration.TestV2WatchKeyInDir
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/v2_http_kv_test.go:880 +0x978
goroutine 1 [chan receive]:
testing.RunTests(0xa47d48, 0xbd5e60, 0x22, 0x22, 0x1)
	/usr/local/go/src/testing/testing.go:556 +0xad6
testing.(*M).Run(0xc2080565f0, 0xbe09e0)
	/usr/local/go/src/testing/testing.go:485 +0x6c
main.main()
	github.com/coreos/etcd/integration/_test/_testmain.go:118 +0x1d5
goroutine 112953 [runnable]:
sync.(*Mutex).Unlock(0xc21c8cd338)
	/usr/local/go/src/sync/mutex.go:82
net/http.(*conn).hijacked(0xc21c8cd2c0, 0x7f0f04604b00)
	/usr/local/go/src/net/http/server.go:133 +0x76
net/http.(*conn).serve(0xc21c8cd2c0)
	/usr/local/go/src/net/http/server.go:1205 +0xb68
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1751 +0x35e
goroutine 112952 [select]:
net/http.(*persistConn).writeLoop(0xc20d9d3080)
	/usr/local/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0xcbc
goroutine 112951 [runnable]:
net/http.(*persistConn).readLoop(0xc20d9d3080)
	/usr/local/go/src/net/http/transport.go:928 +0x9ce
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0xc9f
goroutine 111724 [runnable]:
sync.(*WaitGroup).Wait(0xc216f57160)
	/usr/local/go/src/sync/waitgroup.go:132 +0x169
net/http/httptest.(*Server).Close(0xc216f57130)
	/usr/local/go/src/net/http/httptest/server.go:168 +0x5f
github.com/coreos/etcd/integration.(*member).Terminate(0xc20da14000, 0xc20e3fd290)
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_test.go:822 +0xb6
github.com/coreos/etcd/integration.(*cluster).Terminate(0xc20e3faf20, 0xc20e3fd290)
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_test.go:531 +0x73
runtime.Goexit()
	/usr/local/go/src/runtime/panic.go:288 +0xf6
testing.(*common).FailNow(0xc20e3fd290)
	/usr/local/go/src/testing/testing.go:329 +0x3d
testing.(*common).Fatal(0xc20e3fd290, 0xc20e95de50, 0x1, 0x1)
	/usr/local/go/src/testing/testing.go:364 +0x76
github.com/coreos/etcd/integration.TestV2WatchKeyInDir(0xc20e3fd290)
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/v2_http_kv_test.go:888 +0xe60
testing.tRunner(0xc20e3fd290, 0xbd6148)
	/usr/local/go/src/testing/testing.go:447 +0xbf
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:555 +0xa8b
exit status 2
FAIL	github.com/coreos/etcd/integration	310.688s

--- a/integration/v2_http_kv_test.go
+++ b/integration/v2_http_kv_test.go
@@ -182,7 +182,10 @@ func TestV2CreateUpdate(t *testing.T) {
 	}
 
 	for i, tt := range tests {
-		resp, _ := tc.PutForm(fmt.Sprintf("%s%s", u, tt.relativeURL), tt.value)
+		resp, err := tc.PutForm(fmt.Sprintf("%s%s", u, tt.relativeURL), tt.value)
+		if err != nil {
+			t.Fatalf("#%d: put err = %v, want nil", i, err)
+		}
 		if resp.StatusCode != tt.wStatus {
 			t.Errorf("#%d: status = %d, want %d", i, resp.StatusCode, tt.wStatus)
 		}
@@ -307,7 +310,10 @@ func TestV2CAS(t *testing.T) {
 	}
 
 	for i, tt := range tests {
-		resp, _ := tc.PutForm(fmt.Sprintf("%s%s", u, tt.relativeURL), tt.value)
+		resp, err := tc.PutForm(fmt.Sprintf("%s%s", u, tt.relativeURL), tt.value)
+		if err != nil {
+			t.Fatalf("#%d: put err = %v, want nil", i, err)
+		}
 		if resp.StatusCode != tt.wStatus {
 			t.Errorf("#%d: status = %d, want %d", i, resp.StatusCode, tt.wStatus)
 		}
@@ -403,7 +409,10 @@ func TestV2Delete(t *testing.T) {
 	}
 
 	for i, tt := range tests {
-		resp, _ := tc.DeleteForm(fmt.Sprintf("%s%s", u, tt.relativeURL), nil)
+		resp, err := tc.DeleteForm(fmt.Sprintf("%s%s", u, tt.relativeURL), nil)
+		if err != nil {
+			t.Fatalf("#%d: delete err = %v, want nil", i, err)
+		}
 		if resp.StatusCode != tt.wStatus {
 			t.Errorf("#%d: status = %d, want %d", i, resp.StatusCode, tt.wStatus)
 		}
@@ -499,7 +508,10 @@ func TestV2CAD(t *testing.T) {
 	}
 
 	for i, tt := range tests {
-		resp, _ := tc.DeleteForm(fmt.Sprintf("%s%s", u, tt.relativeURL), nil)
+		resp, err := tc.DeleteForm(fmt.Sprintf("%s%s", u, tt.relativeURL), nil)
+		if err != nil {
+			t.Fatalf("#%d: delete err = %v, want nil", i, err)
+		}
 		if resp.StatusCode != tt.wStatus {
 			t.Errorf("#%d: status = %d, want %d", i, resp.StatusCode, tt.wStatus)
 		}
@@ -562,7 +574,10 @@ func TestV2Unique(t *testing.T) {
 	}
 
 	for i, tt := range tests {
-		resp, _ := tc.PostForm(fmt.Sprintf("%s%s", u, tt.relativeURL), tt.value)
+		resp, err := tc.PostForm(fmt.Sprintf("%s%s", u, tt.relativeURL), tt.value)
+		if err != nil {
+			t.Fatalf("#%d: post err = %v, want nil", i, err)
+		}
 		if resp.StatusCode != tt.wStatus {
 			t.Errorf("#%d: status = %d, want %d", i, resp.StatusCode, tt.wStatus)
 		}
@@ -653,7 +668,10 @@ func TestV2Get(t *testing.T) {
 	}
 
 	for i, tt := range tests {
-		resp, _ := tc.Get(fmt.Sprintf("%s%s", u, tt.relativeURL))
+		resp, err := tc.Get(fmt.Sprintf("%s%s", u, tt.relativeURL))
+		if err != nil {
+			t.Fatalf("#%d: get err = %v, want nil", i, err)
+		}
 		if resp.StatusCode != tt.wStatus {
 			t.Errorf("#%d: status = %d, want %d", i, resp.StatusCode, tt.wStatus)
 		}
@@ -747,7 +765,10 @@ func TestV2QuorumGet(t *testing.T) {
 	}
 
 	for i, tt := range tests {
-		resp, _ := tc.Get(fmt.Sprintf("%s%s", u, tt.relativeURL))
+		resp, err := tc.Get(fmt.Sprintf("%s%s", u, tt.relativeURL))
+		if err != nil {
+			t.Fatalf("#%d: get err = %v, want nil", i, err)
+		}
 		if resp.StatusCode != tt.wStatus {
 			t.Errorf("#%d: status = %d, want %d", i, resp.StatusCode, tt.wStatus)
 		}
@@ -768,12 +789,18 @@ func TestV2Watch(t *testing.T) {
 	u := cl.URL(0)
 	tc := NewTestClient()
 
-	watchResp, _ := tc.Get(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar?wait=true"))
+	watchResp, err := tc.Get(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar?wait=true"))
+	if err != nil {
+		t.Fatalf("watch err = %v, want nil", err)
+	}
 
 	// Set a value.
 	v := url.Values{}
 	v.Set("value", "XXX")
-	resp, _ := tc.PutForm(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar"), v)
+	resp, err := tc.PutForm(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar"), v)
+	if err != nil {
+		t.Fatalf("put err = %v, want nil", err)
+	}
 	resp.Body.Close()
 
 	body := tc.ReadBodyJSON(watchResp)
@@ -802,7 +829,10 @@ func TestV2WatchWithIndex(t *testing.T) {
 	var body map[string]interface{}
 	c := make(chan bool, 1)
 	go func() {
-		resp, _ := tc.Get(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar?wait=true&waitIndex=5"))
+		resp, err := tc.Get(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar?wait=true&waitIndex=5"))
+		if err != nil {
+			t.Fatalf("watch err = %v, want nil", err)
+		}
 		body = tc.ReadBodyJSON(resp)
 		c <- true
 	}()
@@ -816,7 +846,10 @@ func TestV2WatchWithIndex(t *testing.T) {
 	// Set a value (before given index).
 	v := url.Values{}
 	v.Set("value", "XXX")
-	resp, _ := tc.PutForm(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar"), v)
+	resp, err := tc.PutForm(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar"), v)
+	if err != nil {
+		t.Fatalf("put err = %v, want nil", err)
+	}
 	resp.Body.Close()
 
 	select {
@@ -826,7 +859,10 @@ func TestV2WatchWithIndex(t *testing.T) {
 	}
 
 	// Set a value (before given index).
-	resp, _ = tc.PutForm(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar"), v)
+	resp, err = tc.PutForm(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar"), v)
+	if err != nil {
+		t.Fatalf("put err = %v, want nil", err)
+	}
 	resp.Body.Close()
 
 	select {
@@ -863,18 +899,27 @@ func TestV2WatchKeyInDir(t *testing.T) {
 	v := url.Values{}
 	v.Set("dir", "true")
 	v.Set("ttl", "1")
-	resp, _ := tc.PutForm(fmt.Sprintf("%s%s", u, "/v2/keys/keyindir"), v)
+	resp, err := tc.PutForm(fmt.Sprintf("%s%s", u, "/v2/keys/keyindir"), v)
+	if err != nil {
+		t.Fatalf("put err = %v, want nil", err)
+	}
 	resp.Body.Close()
 
 	// Create a permanent node within the directory
 	v = url.Values{}
 	v.Set("value", "XXX")
-	resp, _ = tc.PutForm(fmt.Sprintf("%s%s", u, "/v2/keys/keyindir/bar"), v)
+	resp, err = tc.PutForm(fmt.Sprintf("%s%s", u, "/v2/keys/keyindir/bar"), v)
+	if err != nil {
+		t.Fatalf("put err = %v, want nil", err)
+	}
 	resp.Body.Close()
 
 	go func() {
 		// Expect a notification when watching the node
-		resp, _ := tc.Get(fmt.Sprintf("%s%s", u, "/v2/keys/keyindir/bar?wait=true"))
+		resp, err := tc.Get(fmt.Sprintf("%s%s", u, "/v2/keys/keyindir/bar?wait=true"))
+		if err != nil {
+			t.Fatalf("watch err = %v, want nil", err)
+		}
 		body = tc.ReadBodyJSON(resp)
 		c <- true
 	}()
@@ -910,7 +955,10 @@ func TestV2Head(t *testing.T) {
 	v := url.Values{}
 	v.Set("value", "XXX")
 	fullURL := fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar")
-	resp, _ := tc.Head(fullURL)
+	resp, err := tc.Head(fullURL)
+	if err != nil {
+		t.Fatalf("head err = %v, want nil", err)
+	}
 	resp.Body.Close()
 	if resp.StatusCode != http.StatusNotFound {
 		t.Errorf("status = %d, want %d", resp.StatusCode, http.StatusNotFound)
@@ -919,10 +967,16 @@ func TestV2Head(t *testing.T) {
 		t.Errorf("ContentLength = %d, want > 0", resp.ContentLength)
 	}
 
-	resp, _ = tc.PutForm(fullURL, v)
+	resp, err = tc.PutForm(fullURL, v)
+	if err != nil {
+		t.Fatalf("put err = %v, want nil", err)
+	}
 	resp.Body.Close()
 
-	resp, _ = tc.Head(fullURL)
+	resp, err = tc.Head(fullURL)
+	if err != nil {
+		t.Fatalf("head err = %v, want nil", err)
+	}
 	resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
 		t.Errorf("status = %d, want %d", resp.StatusCode, http.StatusOK)

